api/views/session_views.py

```diff
@@ -187,6 +187,9 @@ def get(self, request, quiz_id):
                 "start_time": session.start_time.isoformat() if session.start_time else None,
                 "end_time": session.start_time.isoformat() if session.end_time else None,
                 "code": session.code,
+                "num_of_participants": QuizSessionStudent.objects.filter(
+                    quiz_session__code=session.code
+                ).count(),
```

Comment:
You could use `session.students.count()` here.

Having a `related_field` in `QuizSessionStudent ` allows you to do that:

```
class QuizSessionStudent(models.Model):
    username = models.CharField(max_length=200)
    joined_at = models.DateTimeField(default=timezone.now)
    quiz_session = models.ForeignKey(QuizSession, on_delete=models.CASCADE, related_name="students")
    score = models.IntegerField(default=0)
    skip_count = models.IntegerField(default=0)

    class Meta:
        db_table = "api_quiz_session_student"
```

Related fields allow you to go the other way around.

Reply:
Made this change